api/src/controllers/users.ts

```diff
@@ -332,7 +332,15 @@ router.post(
 			throw new InvalidCredentialsError();
 		}
 
-		if (!req.body.password) {
+		const usersService = new UsersService({
+			accountability: req.accountability,
+			schema: req.schema,
+		});
+
+		const currentUser = await usersService.readOne(req.accountability.user, { fields: ['provider'] });
```

Comment:
```suggestion
		const currentUser = await getDatabase()
			.select('provider')
			.from('directus_users')
			.where({ id: req.accountability.user })
			.first();
```
We should call the database here directly because under the "minimal app permissions" users themselves do not have read access to their own `provider` field. This results in the following error when trying to set up 2fa

<img width="653" height="629" alt="image" src="https://github.com/user-attachments/assets/2bc1e980-a41b-4fd5-8daa-c9689c92358e" />